### PR TITLE
fix: [CDS-41104]: Sync step

### PR DESCRIPTION
**Pull request by satyam-harness**
#### Screenshots
image

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Comment by satyam-harness** [comment]
retrigger titlecheck

**Comment by satyam-harness** [comment]
retrigger cypress-pipeline

File: src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/types.tsx [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+
+export enum POLICY_OPTIONS {
+  FOREGROUND = 'foreground',
+  BACKGROUND = 'background',
+  ORPHAN = 'orphan'
+}
+
+export const policyOptions = [
+  {
+    label: 'Foreground',
```
Comment (ishantmehtaharness) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/types.tsx:28]: we should use i18n strings here

**Review by ishantmehtaharness — APPROVED** [review]
Looks Good

File: src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStep.tsx [hunk 2]
```diff
@@ -0,0 +1,197 @@
...
+      formikRef,
+      isNewStep,
+      readonly,
+      onChange,
+      allowableTypes,
+      customStepProps
+    } = props
+
+    if (this.isTemplatizedView(stepViewType)) {
+      return <></>
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStep.tsx:173]: Is the input set view yet to be implemented?

File: src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStepBase.tsx [hunk 3]
```diff
@@ -0,0 +1,384 @@
...
+  readonly,
+  allowableTypes,
+  stepViewType
+}: SyncStepFormContentInterface): React.ReactElement {
+  const { getString } = useStrings()
+  const { expressions } = useVariablesExpression()
+  // const { values: formValues } = formik
+  const { accountId, projectIdentifier, orgIdentifier } =
+    useParams<PipelineType<PipelinePathProps & AccountPathProps>>()
+  // const { repoIdentifier, branch } = useQueryParams<GitQueryParams>()
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStepBase.tsx:53]: Please remove commented code, if not needed

File: src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStep.tsx (hunk 2 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStep.tsx:173]: yes

File: src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStepBase.tsx [hunk 4]
```diff
@@ -0,0 +1,384 @@
...
+      accountIdentifier: accountId
+    })
+  }, [filters, accountId, orgIdentifier, projectIdentifier])
+
+  const itemRenderer = memoize((item: SelectOption, itemProps: IItemRendererProps) => (
+    <ItemRendererWithMenuItem
+      item={item}
+      itemProps={itemProps}
+      disabled={loading}
+      // style={imagesListError ? { lineClamp: 1, width: 400, padding: 'small' } : {}}
```
Comment (PratyushGar) [inline src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStepBase.tsx:133]: same here

**Review by Eric761 — APPROVED** [review]

**Review by PratyushGar — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStepBase.tsx (hunk 4 above)
Comment (satyam-harness) [reply src/modules/70-pipeline/components/PipelineSteps/Steps/SyncStep/SyncStepBase.tsx:133]: done

**Pull request merged**